app/javascript/components/widgets/numbers/falling-blocks.jsx

```diff
@@ -27,7 +27,8 @@ class Scene extends React.Component {
       ...currentState,
     };
     this.scene = React.createRef();
-    this.timer = null;
+    this.addBlockTimer = null;
+    this.freezeTimer = null;
```

Comment:
Why iterate on the collection of bodies outside of the render loop? It seems to me a likely culprit could be the matter js builin render method. Excerpt from https://github.com/liabru/matter-js/wiki/Rendering

> There is an included debug renderer called Matter.Render. This module is an optional canvas based renderer for visualising instances of Matter.Engine. It is mostly intended for development and debugging purposes, but may also be suitable starting point for simple games.